wezterm-input-types/src/lib.rs

```diff
@@ -1648,8 +1648,9 @@ impl KeyEvent {
         {
             // Check for simple text generating keys
             match &self.key {
+                // Don't send legacy codes with DEL and ESC
+                Char('\x1b') | Char('\x7f') => {}
```

Comment:
```suggestion
                // Esc should only be passed through as-is if disambiguate_escape_codes is false
                Char('\x1b')  if flags.contains(KittyKeyboardFlags::DISAMBIGUATE_ESCAPE_CODES) => {}
                // In kitty mode, DEL is always sent as the special escape \e[3~
                Char('\x7f') => ()
```

Although... it seems like currently having flags set to `0` might behave basically the same as setting it to `1`. I'm not sure if that is intentional.